packages/protocol/test/AccountRegistry/AccountRegistryManager.js

```diff
@@ -0,0 +1,233 @@
+                const s = signature[2].slice(2);
+                const v = signature[0].slice(-2);
+                const sig = r + s + v;
+
+                await behaviour.registerAZTECExtension(address, linkedPublicKey, spendingPublicKey, sig);
+
+                const storedLinkedPublicKey = await behaviour.accountMapping.call(address);
+                expect(storedLinkedPublicKey).to.equal(linkedPublicKey);
+            });
+
+            it('should upgrade to a new Account Registry behaviour contract', async () => {
+                const testBehaviour = await TestBehaviour.new();
+                const existingProxy = await manager.proxy.call();
+
+                await manager.upgradeAccountRegistry(existingProxy, testBehaviour.address);
+
+                const postUpgradeProxy = await manager.proxy.call();
+                expect(postUpgradeProxy).to.equal(existingProxy);
+
+                const newBehaviourAddress = await manager.getImplementation.call(existingProxy);
+                const expectedNewBehaviourAddress = testBehaviour.address;
+                expect(newBehaviourAddress).to.equal(expectedNewBehaviourAddress);
+
+                const isNewFeatureImplemented = 'newFeature()' in testBehaviour.methods;
+                expect(isNewFeatureImplemented).to.equal(true);
+            });
+
+            it('should keep state after upgrade', async () => {
+                // register an account with the extension, which will be put in storage under accountMapping
+                const { privateKey, address } = secp256k1.generateAccount();
+                const linkedPublicKey = keccak256('0x01');
+                const spendingPublicKey = keccak256('0x0');
+
+                const domain = signer.generateAccountRegistryDomainParams(behaviour.address);
+                const message = {
+                    account: address,
+                    linkedPublicKey,
+                };
+
+                const encodedTypedData = typedData.encodeTypedData({
+                    domain,
+                    ...ACCOUNT_REGISTRY_SIGNATURE,
+                    message,
+                });
+
+                const signature = secp256k1.ecdsa.signMessage(encodedTypedData, privateKey);
+
+                const r = signature[1];
+                const s = signature[2].slice(2);
+                const v = signature[0].slice(-2);
+                const sig = r + s + v;
+
+                await behaviour.registerAZTECExtension(address, linkedPublicKey, spendingPublicKey, sig);
+
+                // perform upgrade, and confirm that registered linkedPublicKey is still present in storage
+                const testBehaviour = await TestBehaviour.new();
+                const preUpgradeProxy = await manager.proxy.call();
+                await manager.upgradeAccountRegistry(preUpgradeProxy, testBehaviour.address);
+
+                const storedLinkedPublicKey = await behaviour.accountMapping.call(address);
```

Comment:
This is completely avoiding the Proxy contract, only testing that the deployed behaviour is setting state (in itself, not in proxy context) correctly.

Reply:
Good point, have updated the tests